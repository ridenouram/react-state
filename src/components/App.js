import React, { PureComponent } from 'react';
import ColorPicker from './ColorPicker';
import ColorDisplay from './ColorDisplay';

export default class App extends PureComponent {
  state = {
    color: ''
  };

  colorClick = color => {
    this.setState({ color });
  }

  render() {
    const { color } = this.state;

    return (
      <>
        <ColorPicker colorFunction={this.colorClick} />
        <ColorDisplay color={ color } />
      </>
    );
  }

  componentDidUpdate() {
    const { color } = this.state;
    console.log(color);
  }
}
