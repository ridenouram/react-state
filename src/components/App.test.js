import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('app component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
  it('changes color', () => {
    const wrapper = shallow(<App />);
    wrapper.instance().colorClick('red');
    expect(wrapper.state('color')).toEqual('red');
  });
});
