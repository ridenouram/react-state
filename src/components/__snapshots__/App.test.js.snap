// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component matches snapshot 1`] = `
<Fragment>
  <ColorPicker
    colorFunction={[Function]}
  />
  <ColorDisplay
    color=""
  />
</Fragment>
`;
